src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -1323,18 +1326,21 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 			this._logService.trace(`Already reconnected, to ${this._reconnectedTerminals?.length} terminals so returning`);
 			return;
 		}
-		this._reconnectedTerminals = this._terminalService.getReconnectedTerminals(ReconnectionType)?.filter(t => !t.isDisposed);
+		this._reconnectedTerminals = [];
+		for (const terminal of this._terminalService.getReconnectedTerminals(ReconnectionType) || []) {
+			const reconnectionData = terminal.shellLaunchConfig.attachPersistentProcess?.reconnectionProperties?.data as IReconnectionTaskData;
+			if (reconnectionData) {
+				this._reconnectedTerminals.push({ ...terminal, reconnectionData });
+			} else {
+				this._logService.trace(`Terminal was owned by tasks but has no reconnection data ${terminal.instanceId}, ${JSON.stringify(terminal.shellLaunchConfig)}`);
```

Comment:
I'm not sure how this was possible before 🤔

Reply:
which makes me think this shouldn't have ever been undefined

https://github.com/microsoft/vscode/pull/182896/files#diff-5d6d50fef79440567acc5f9a977c908f45f59342bb4d5e19ef5b9ee25e02920cL1280